routers/api/actions/artifacts.go

```diff
@@ -0,0 +1,459 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package actions
+
+import (
+	"bytes"
+	"compress/gzip"
+	gocontext "context"
+	"crypto/md5"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"code.gitea.io/gitea/models/actions"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/storage"
+	"code.gitea.io/gitea/modules/web"
+)
+
+const artifactXTfsFileLengthHeader = "x-tfs-filelength"
+const artifactXActionsResultsMD5Header = "x-actions-results-md5"
+
+// const artifactXActionsResultsCRC64Header = "x-actions-results-crc64"
+
+const artifactRouteBase = "/_apis/pipelines/workflows/{taskID}/artifacts"
+
+func ArtifactsRoutes(goctx gocontext.Context, prefix string) *web.Route {
+	m := web.NewRoute()
+	m.Use(withContexter(goctx))
```

Comment:
So how to handle the permission check? Since very runner has his token, we can have a middleware to check that.

Reply:
I add Bearer token auth middleware for now in artifact router